variation-commons-core/src/main/java/uk/ac/ebi/eva/commons/core/models/ws/VariantWithSamplesAndAnnotation.java

```diff
@@ -51,6 +54,22 @@ public VariantWithSamplesAndAnnotation(String chromosome, long start, long end,
         sourceEntries = new HashMap<>();
     }
 
+    public VariantWithSamplesAndAnnotation(Variant variant) {
+        super(variant.getChromosome(), variant.getStart(), variant.getEnd(), variant.getReference(), variant.getAlternate());
+
+        Map<String, VariantSourceEntryWithSampleNames> sourceEntries = new HashMap<>();
```

Comment:
This could be assigned directly to the class attribute instead of creating a new variable.